services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocument.java

```diff
@@ -15,76 +15,95 @@
- * to DynamoDB using EnhancedDynamoDB client.
- *
- * TODO : Add some examples in the Java Doc after API Surface review.
+ * Interface representing the Document API for DynamoDB. The Document API operations are designed to work with open content,
+ * such as data with no fixed schema, data that cannot be modeled using rigid types, or data that has a schema.
+ * This interface provides all the methods required to access a Document, as well as constructor methods for creating a
+ * Document that can be used to read and write to DynamoDB using the EnhancedDynamoDB client.
+ * Additionally, this interface provides flexibility when working with data, as it allows you to work with data that is not
+ * necessarily tied to a specific data model.
+ * The EnhancedDocument interface provides two ways to use AttributeConverterProviders:
+ * <p>Enhanced Document with default attribute Converter to convert the attribute of DDB item to basic default primitive types in
+ * Java
+ * {@snippet :
+ * EnhancedDocument enhancedDocument = EnhancedDocument.builder().attributeConverterProviders(AttributeConverterProvider
+ * .defaultProvider()).build();
+ *}
+ * <p>Enhanced Document with Custom attribute Converter to convert the attribute of DDB Item to Custom Type.
+ * {@snippet :
+ * // CustomAttributeConverterProvider.create() is an example for some Custom converter provider
+ * EnhancedDocument enhancedDocumentWithCustomConverter = EnhancedDocument.builder().attributeConverterProviders
+ * (CustomAttributeConverterProvider.create(), AttributeConverterProvide.defaultProvider())
+ * .putWithType("customObject", customObject, EnhancedType.of(CustomClass.class))
+ * .build();
+ *}
+ * <p>Enhanced Document can be created with Json as input using Static factory method.In this case it used
+ * defaultConverterProviders.
+ * {@snippet :
+ * EnhancedDocument enhancedDocumentWithCustomConverter = EnhancedDocument.fromJson("{\"k\":\"v\"}");
+ *}
+ * The attribute converter are always required to be provided, thus for default conversion
+ * {@link AttributeConverterProvider#defaultProvider()} must be supplied.
  */
 @SdkPublicApi
 public interface EnhancedDocument {
 
     /**
-     * Convenience factory method - instantiates an <code>EnhancedDocument</code> from the given JSON String.
-     *
-     * @param json The JSON string representation of DynamoDB Item.
+     * Creates a new <code>EnhancedDocument</code> instance from a JSON string.
+     * The {@link AttributeConverterProvider#defaultProvider()} is used as the default ConverterProvider.
+     * To use a custom ConverterProvider, use the builder methods: {@link Builder#json(String)} to supply the JSON string,
+     * then use {@link Builder#attributeConverterProviders(AttributeConverterProvider...)} to provide the custom
+     * ConverterProvider.
+     * {@snippet :
+     * {@code EnhancedDocument documentFromJson = EnhancedDocument.fromJson("{\"key\": \"Value\"}");}
```

Comment:
Haven't played with `@snippet` yet, do you need the inner `@code` here?

Reply:
With JDK 18 generated Java docs we started using this and prefer @snippet over  @code @zoewangg correct me if I am wrong.

Reply:
yeah what I mean is, you have both here. Since you have `@snippet`, I think `@code` is unnecessary right?

Reply:
oh I got it now , will remove it